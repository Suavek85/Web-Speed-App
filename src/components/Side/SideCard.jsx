import React from 'react'
import ReactTooltip from "react-tooltip"
import classNames from 'classnames/bind'
import { useMediaQuery } from 'react-responsive'
import { useSelector } from "react-redux"
import { appStates } from '../../constants/states'
import { desktopWidth } from'../../scss/mediaqueries'
import SkeletonCircle from '../Skeleton/SkeletonCircle'
import InfoCircle from '../svgs/InfoCircle/InfoCircle'
import SkeletonText from '../Skeleton/SkeletonText'
import LightBulbUI from '../svgs/LightBulb/LightBulbUI'
import ScoreCircle from '../svgs/ScoreCircle/ScoreCircle'
import styles from './SideCard.scss'


export default function Section(props) {

  const { content, data, position } = props
  const { header, mainContent, result } = content 

  const getAppState = useSelector(state => state.stateReducer.getAppState)
  const loadingStatus = getAppState === appStates.LOADING
  const isDesktop = useMediaQuery(desktopWidth)

  //STYLES
  const cx = classNames.bind(styles)
  const getClass = cx({
    block: true,
    blockGeneric: position === 'generic',
    blockTop: position === 'top',
    blockBottom: position === 'bottom',
  })

  const getColorClass = arg => cx({
    blockContentAlert: arg === 'SLOW',
    blockContentInfo: arg === 'FAST',
    blockContentWarning: arg === 'AVERAGE',
  })

  const getDescrptionWrapperClass = cx({
    blockDesc: true,
    blockGenericDesc: position === 'generic',
    blockResultDesc: position !== 'generic',
  })
  

  // GENERIC ELEMENTS

  const genericDescription = (
    <p>{ mainContent }</p>
  )

  const genericPara = (el, index) => (
    <p key={index}>{ result[index]  } 
      <span style={{ marginLeft:'5px' }} className={ getColorClass(el)}>
        { el !== '?' && el }
        { el === '?' && loadingStatus && <SkeletonText />}
      </span>
    </p>
  )

  //RESULTS ELEMENTS

  const description = (
    <>
      <span style={{marginLeft: '7px'}}>
        <InfoCircle mainContent={mainContent} />
      </span>
      <ReactTooltip id={ mainContent } type='info'>
        <span>{ mainContent }</span>
      </ReactTooltip>
    </>
  )

  const scorePara = (el, index) => { 
    return el !== '?' && !loadingStatus && (
      <p key={index}>
        Score: 
        <span>
          <ScoreCircle score={el} />
        </span>
      </p>
    )
  }

  const scoreParaLoading = (el, index) => { 
    return el === '?' && loadingStatus && (
      <p className={scorePara} key={index}>
        Score: 
        <span>
          <SkeletonCircle />
        </span>
      </p>
    )
  }

  return (
    <div className={ getClass } >
      { isDesktop && position === 'generic' && <LightBulbUI /> }
      <div className={ styles.blockContent }>
        <div className={ getDescrptionWrapperClass }>
          <h2>
            { header }
          </h2>
          { position === 'generic' && genericDescription }
          { position !== 'generic' && description}
        </div>
        { position === 'generic' && data.map(genericPara) }
        { position !== 'generic' && data.map(scorePara) }
        { position !== 'generic' && data.map(scoreParaLoading) }
      </div>
    </div>
  )
}
