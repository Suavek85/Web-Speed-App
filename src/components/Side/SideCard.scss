@import '../../scss/styles';

$block: sidecard;
$blockGeneric: modifier($block, generic);
$blockTop: modifier($block, top);
$blockBottom: modifier($block, bottom);
$blockContent: elem($block, content);
$blockContentAlert: modifier($blockContent, alert);
$blockContentWarning: modifier($blockContent, warning);
$blockContentInfo: modifier($blockContent, info);
$blockDesc: elem($block, desc);
$blockGenericDesc: modifier($blockDesc, generic);
$blockResultDesc: modifier($blockDesc, result);
$blockBulb: elem($block, bulb);
$blockScore: elem($block, score);

.#{$block} {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  background-color: var(--color-inner-background);

  &--generic {
    border-bottom: 1px solid var(--color-main-background);
  }

  &--top {
    border-bottom: 1px solid var(--color-main-background);
  }

  &--bottom {
    border-bottom: 1px solid var(--color-main-background);
  }
}

.#{$blockContent} {
  position: relative;
  display: flex;
  flex-direction: column;
  padding: 0 30px;

  > div > h2 {
    margin-bottom: 0;
    font-size: em(17);
    color: var(--color-header-font);
  }

  p {
    display: flex;
    flex-direction: row;
    align-items: center;
    margin-top: 10px;
    font-size: em(13);
    color: var(--color-very-small-font);
  }

  p span {
    font-weight: 500;
  }

  &--alert {
    color: $color-alert;
  }

  &--warning {
    color: $color-warning-loading;
  }

  &--info {
    color: $color-green-success;
  }
}

.#{$blockDesc} {
  display: flex;
  margin-bottom: 5px;

  &--generic {
    align-items: flex-end;
  }

  &--result {
    align-items: flex-end;
  }
}

.#{$blockBulb} {
  position: absolute;
  top: 5px;
  right: 5px;
}

:export {
  block: $block;
  blockGeneric: $blockGeneric;
  blockTop: $blockTop;
  blockBottom: $blockBottom;
  blockContent: $blockContent;
  blockContentAlert: $blockContentAlert;
  blockContentWarning: $blockContentWarning;
  blockContentInfo: $blockContentInfo;
  blockDesc: $blockDesc;
  blockGenericDesc: $blockGenericDesc;
  blockResultDesc: $blockResultDesc;
  blockBulb: $blockBulb;
}
