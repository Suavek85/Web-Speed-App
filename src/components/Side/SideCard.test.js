import React from 'react'
import * as redux from 'react-redux'
import { shallow } from 'enzyme'
import SideCard from './SideCard'

describe('SideCard snapshot', () => {

  it('should match the snapshot', () => {
    const spyUseSelector = jest.spyOn(redux, 'useSelector')
    spyUseSelector.mockReturnValue('INACTIVE')
    const wrapper = shallow(<SideCard/>)
    expect(wrapper).toMatchSnapshot()
  })

})