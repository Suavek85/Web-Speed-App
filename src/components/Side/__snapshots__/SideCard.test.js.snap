// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideCard snapshot should match the snapshot 1`] = `
<div
  className="block"
>
  <div>
    <div
      className="blockDesc blockResultDesc"
    >
      <h2 />
      <span
        style={
          Object {
            "marginLeft": "7px",
          }
        }
      >
        <Section
          mainContent={null}
        />
      </span>
      <ReactTooltip
        clickable={false}
        id={null}
        insecure={true}
        resizeHide={true}
        type="info"
        wrapper="div"
      >
        <span />
      </ReactTooltip>
    </div>
  </div>
</div>
`;
